x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/utils.test.ts

```diff
@@ -68,39 +73,103 @@ describe('crawlerDomainServerToClient', () => {
   });
 });
 
+describe('crawlRequestServerToClient', () => {
+  it('converts the API payload into properties matching our code style', () => {
```

Comment:
[Not a change request, just rambling] I've mentioned this to Jason before, but I wish we'd put our snake_case->camelCase transmogs in the `server/` folder and not in `client/`. I'd rather all our frontend code only have to handle correctly camelCased vars and that we use our `server/` code as a light middleware layer that fixes/cleans up data coming in from our backend.

When I last yarned with @JasonStoltz, I believe he preferred our server routes code be straight pass-throughs. Personally I like the idea of consolidating some of our backend cruft to one place, but I see the architectural tradeoff (e.g. cleaner but harder to find code). I wonder if we could do something like a generic `convertSnakeCaseToCamelCase()` helper that we apply by default to App Search API JSON responses.

Mostly just writing this here to drop my 2c and plant an idea in case someone likes it in the future

Reply:
I'm not sure I like the middleware idea, I'll think about it tho

Reply:
No worries then if neither of you are fans!